src/core/search/search.cc

```diff
@@ -188,13 +195,18 @@ struct BasicSearch {
     if (sub_results.empty())
       return vector<DocId>{};
 
-    // Unifying from smallest to largest is more efficient.
-    // AND: the result only shrinks, so starting with the smallest is most optimal.
-    // OR: unifying smaller sets first reduces the number of element traversals on average.
+      // Unifying from smallest to largest is more efficient.
+      // AND: the result only shrinks, so starting with the smallest is most optimal.
+      // OR: unifying smaller sets first reduces the number of element traversals on average.
+
+      // Ignore Wmaybe-uninitialized because it yields maybe-uninitialized with GCC 12
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
     sort(sub_results.begin(), sub_results.end(),
          [](const auto& l, const auto& r) { return l.Size() < r.Size(); });
+#pragma GCC diagnostic pop
 
```

Comment:
The clang build from the CI passes successfully. I had to add those pragmas because GCC wrongfully emits a warning here.

Reply:
@royjacobson 

It didn't help. Actually it happened once (3/4 built successfully), but now it fails again 😵 

You can clearly see its those lines because of the sort call + lambda 
```
inlined from 'void std::__insertion_sort(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with 
_RandomAccessIterator = __gnu_cxx::__normal_iterator<dfly::search::{anonymous}::IndexResult*, 
vector<dfly::search::{anonymous}::IndexResult> >; _Compare = __gnu_cxx::__ops::_Iter_comp_iter<dfly::search::
{anonymous}::BasicSearch::UnifyResults(std::vector<dfly::search::{anonymous}::IndexResult>&&, LogicOp)::
<lambda(const auto:34&, const auto:35&)> >]' at /usr/include/c++/12.2.1/bits/stl_algo.h:1814:17:
```

Reply:
Which variable does GCC think is unused?

Reply:
Not unused, it sees a uninitialized use

Reply:
a member of std::vector

We concluded with Roy that its a wrong warning

Reply:
I applied the pragma to the whole anonmyous namespace and it seems to have fixed it